### DISCUSSION TITLE
How to disable that new HDR function?

### DISCUSSION THREAD

**Discussion by pdmk** [iina/iina#3781] [Q&A]
IINA was updated today to 1.3.0 version 

I don't have HDR monitor, previously IINA somehow converted all HDR videos into SDR

Now it forces HDR,so I have to manually turn off the HDR via option every time I start video which was encoded in HDR. Otherwise the video is unwatchable

Please add option to permanently turn it on/off in menu, because turning it off every time is really annoying  

thank you

**Comment by low-batt** [comment]
I'm hearing:
- IINA is enabling HDR for a non-HDR display. A defect.
- IINA needs to provide a way to turn HDR off that persists. A feature request.

Best to enter issues for these. For the defect we definitely need details about the display and the video being played. Please enable logs in Preferences/Advanced and include the logs in the issue about the defect.

I didn't work on the HDR feature, so I don't have a guess as to what is going wrong.

**Reply by pdmk** [reply]
> 

sorry for late reply
This applies to all HDR videos, it doesn't apply to videos that are not encoded as HDR (the function doesn't turn on automatically)

And I am using iMac 27" 2017, but I assume it applies to any non-HDR screens.

Everything was ok in previous version

attached are the logs. 

Again, the very simple solution would be the option to permanently turn it on/off in the settings. 

thank you
[iina.log](https://github.com/iina/iina/files/8851917/iina.log)
[mpv.log](https://github.com/iina/iina/files/8851918/mpv.log)

**Comment by CarterLi** [comment]
IINA tests [`window?.screen?.maximumPotentialExtendedDynamicRangeColorComponentValue`](https://developer.apple.com/documentation/appkit/nsscreen/3180381-maximumpotentialextendeddynamicr).

According to [the doc](https://developer.apple.com/documentation/appkit/nsscreen/3180381-maximumpotentialextendeddynamicr#discussion), `maximumPotentialExtendedDynamicRangeColorComponentValue > 1.0` indicates that the screen supports EDR values.

Note old Intel MBP does support EDR mode ( maximum luminance 500 nits ). The image looks bad because IINA still lacks automatic tone mapping.

**Reply by CarterLi** [reply]
If someone is able to get the maximum luminance of user's screen somehow, automatic tone mapping is not hard to do.

**Reply by svobs** [reply]
> If someone is able to get the maximum luminance of user's screen somehow, automatic tone mapping is not hard to do.

@CarterLi I know almost nothing about this stuff, so I don't know if this is a great idea, but just trying to help out if possible.

Assuming the maximum luminance of a screen isn't known but was given by the user (maybe via a number entry box or slider), would you be able to feed that into an equation or algorithm which could produce pretty good results? I'm not sure that if screen makers provided a number, that would even be guaranteed to be honest or accurate. But it shouldn't be too hard to get a unique identifier for each of the user's screens, so maybe they could do a one-time calibration. If so, I could probably help with some of the work on that. Or would there be a lot of other factors to control for?

**Reply by CarterLi** [reply]
You may test it now with

```
--target-peak=500
--tone-mapping=hable ( or something else if it looks better )
```

**Reply by CarterLi** [reply]
<https://mpv.io/manual/stable/#options-target-peak>

**Comment by low-batt** [comment]
I have created issue #3807 for this discussion.
